Sources/web3swift/Transaction/EIP2930Envelope.swift

```diff
@@ -0,0 +1,412 @@
+        self.s = BigUInt(sData)
+
+        // swiftlint:disable force_unwrapping
+        switch rlpItem[RlpKey.destination.rawValue]!.content {
+            // swiftlint:enable force_unwrapping
+        case .noItem:
+            self.to = EthereumAddress.contractDeploymentAddress()
+        case .data(let addressData):
+            if addressData.count == 0 {
+                self.to = EthereumAddress.contractDeploymentAddress()
+            } else if addressData.count == 20 {
+                guard let addr = EthereumAddress(addressData) else { return nil }
+                self.to = addr
+            } else { return nil }
+        case .list:
+            return nil
+        }
+
+        // swiftlint:disable force_unwrapping
+        switch rlpItem[RlpKey.accessList.rawValue]!.content {
+            // swiftlint:enable force_unwrapping
+        case .noItem:
+            self.accessList = []
+        case .data:
+            return nil
+        case .list:
+            // decode the list here
+            // swiftlint:disable force_unwrapping
+            let accessData = rlpItem[RlpKey.accessList.rawValue]!
+            // swiftlint:enable force_unwrapping
+            let itemCount = accessData.count ?? 0
+            var newList: [AccessListEntry] = []
+            for index in 0...(itemCount - 1) {
+                guard let itemData = accessData[index] else { return nil }
+                guard let newItem = AccessListEntry(rlpItem: itemData)  else { return nil }
+                newList.append(newItem)
+            }
+            self.accessList = newList
+        }
+    }
+
+    public init(to: EthereumAddress, nonce: BigUInt? = nil,
+                chainID: BigUInt? = nil, value: BigUInt? = nil, data: Data,
+                v: BigUInt = 1, r: BigUInt = 0, s: BigUInt = 0,
+                options: TransactionOptions? = nil) {
+        self.to = to
+        self.nonce = nonce ?? options?.resolveNonce(0) ?? 0
+        self.internalChainID = chainID ?? 0
+        self.value = value ?? options?.value ?? 0
+        self.data = data
+        self.v = v
+        self.r = r
+        self.s = s
+        // decode gas options, if present
+        self.gasPrice = options?.resolveGasPrice(0) ?? 0
+        self.gasLimit = options?.resolveGasLimit(0) ?? 0
+        self.accessList = options?.accessList ?? []
+    }
+
+    // memberwise
+    public init(to: EthereumAddress, nonce: BigUInt = 0,
+                chainID: BigUInt = 0, value: BigUInt = 0, data: Data,
+                gasPrice: BigUInt = 0, gasLimit: BigUInt = 0, accessList: [AccessListEntry]? = nil,
+                v: BigUInt = 1, r: BigUInt = 0, s: BigUInt = 0) {
+        self.to = to
+        self.nonce = nonce
+        self.internalChainID = chainID
+        self.value = value
+        self.data = data
+        self.gasPrice = gasPrice
+        self.gasLimit = gasLimit
+        self.accessList = accessList ?? []
+        self.v = v
+        self.r = r
+        self.s = s
+    }
```

Comment:
The first init is defined/required by the protocol. As I said in my reply in AbstractEnvelope, the nil's allow us to chain down to getting the value from the passed options, or a default. The order of precedence is `direct parameter > options > default value`

The second init is the true type-specific memberwise init, with appropriate defaults applied.

The first cannot be removed and conform to AbstractEnvelope, and would break EnvelopeFactory as a result. The second can be removed, but I thought it would be good to keep in case someone decides to create the envelope manually in their code for some reason. In fact EthereumTransaction uses the LegacyEnvelope version of this init in the shim layer. We could add shims for the new types as well, if desired.

Reply:
Oh, speaking about that optional to non optional assigning i see why this works. You've hide it into object implementation.

Although i'm ok with the chain that you're making here i disagree with the tools that you're using for it. But i'll continue my point in `AbstractEnvelop` discussion coz it's more relative there.

Reply:
I saw your point there, and removed the default implementation from `AbstractEnvelope`